imap/monitor.go

```diff
@@ -147,41 +150,46 @@ func checkForNewEmails(im models.IMAP) {
 				}
 			}
 
-			body := string(append(m.Email.Text, m.Email.HTML...)) // Not sure if we need to check the Text as well as the HTML. Perhaps sometimes Text only emails won't have an HTML component?
-			rid := goPhishRegex.FindString(body)
+			rids, err := checkRIDs(m.Email) // Search email Text, HTML, and each attachment for rid parameters
 
-			if rid != "" {
-				rid = rid[5:]
-				log.Infof("User '%s' reported email with rid %s", m.Email.From, rid)
-				result, err := models.GetResult(rid)
-				if err != nil {
-					log.Error("Error reporting GoPhish email with rid ", rid, ": ", err.Error())
-					reportingFailed = append(reportingFailed, m.UID)
-				} else {
-					err = result.HandleEmailReport(models.EventDetails{})
+			if err != nil {
+				log.Errorf("Error searching email for rids from user '%s': %s", m.Email.From, err.Error())
+			} else {
+				if len(rids) < 1 {
+					// In the future this should be an alert in Gophish
+					log.Infof("User '%s' reported email with subject '%s'. This is not a GoPhish campaign; you should investigate it.", m.Email.From, m.Email.Subject)
+				}
+				for rid := range rids {
+					log.Infof("User '%s' reported email with rid %s", m.Email.From, rid)
+					result, err := models.GetResult(rid)
 					if err != nil {
-						log.Error("Error updating GoPhish email with rid ", rid, ": ", err.Error())
+						log.Error("Error reporting GoPhish email with rid ", rid, ": ", err.Error())
+						reportingFailed = append(reportingFailed, m.SeqNum)
```

Comment:
Should we just `continue` here and bump the `else` up a level?

Reply:
Done.